### ISSUE TITLE
Possible bug in PatchNCELoss

### ISSUE THREAD

**Issue by raimannma** [taesungp/contrastive-unpaired-translation#47]
```
# diagonal entries are similarity between same features, and hence meaningless.
# just fill the diagonal with very small number, which is exp(-10) and almost zero
diagonal = torch.eye(npatches, device=feat_q.device, dtype=self.mask_dtype)[None, :, :]
l_neg_curbatch.masked_fill_(diagonal, -10.0)
l_neg = l_neg_curbatch.view(-1, npatches)
```
[Link to the Code](https://github.com/taesungp/contrastive-unpaired-translation/blob/master/models/patchnce.py#L46)

The comment says that you want to put a very small number exp(-10) in the diagonal, but later you set -10.0 as value.

So either the comment is wrong or the exponential function is missing in the code below.
Or did I miss something?

**Comment by taesungp** [comment]
exp is implicitly applied in the Cross Entropy Loss ([here](https://github.com/taesungp/contrastive-unpaired-translation/blob/master/models/patchnce.py#L51), [definition](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html#torch.nn.CrossEntropyLoss)).

**Comment by raimannma** [comment]
Ok, thanks.
